core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -959,7 +974,8 @@ public Weighting getWeightingForCH( WeightingMap weightingMap, FlagEncoder encod
         // get requested weighting name
         String weightingStr = weightingMap.getWeighting().toLowerCase();
         if (weightingStr.isEmpty())
-            weightingStr = chWeightingStr;
+            // TODO Ugly workaround or valid?
+            weightingStr = getCHWeighting().get(0);
```

Comment:
This is valid as we always assume at least 'fastest' in the weighting list. Maybe throw an exception here if it is empty with a more clear exception than the ArrayOutOfBounds thing